main.tf

```diff
@@ -357,6 +358,277 @@ resource "aws_s3_bucket_object_lock_configuration" "default" {
+  count = local.malware_protection
+  role  = module.s3_malware_protection_role[0].arn
+
+  protected_resource {
+    s3_bucket {
+      bucket_name     = aws_s3_bucket.default.id
+      object_prefixes = try(var.malware_protection.object_prefixes, null)
+    }
+  }
+
+  actions {
+    tagging {
+      status = "ENABLED"
+    }
+  }
+
+}
+
+data "aws_iam_policy_document" "s3_bucket_policy" {
+  count = local.malware_protection
+  statement {
+    sid    = "NoReadExceptForClean"
+    effect = "Deny"
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+    actions = [
+      "s3:GetObject",
+      "s3:GetObjectVersion"
+    ]
+    resources = [
+      "arn:aws:s3:::${aws_s3_bucket.default.id}",
+      "arn:aws:s3:::${aws_s3_bucket.default.id}/*"
+    ]
+    condition {
+      test     = "StringNotEquals"
+      variable = "s3:ExistingObjectTag/GuardDutyMalwareScanStatus"
+      values   = ["NO_THREATS_FOUND"]
+    }
+    condition {
+      test     = "ForAnyValue:ArnNotEquals"
+      variable = "aws:PrincipalArn"
+      values = [
+        "arn:aws:iam::${local.account_id}:root",
+        "arn:aws:iam::${local.account_id}:assumed-role/${local.malware_iam_role_name}Role/GuardDutyMalwareProtection",
+        module.s3_malware_protection_role[0].arn
+      ]
+    }
+  }
+
+  statement {
+    sid    = "OnlyGuardDutyCanTag"
+    effect = "Deny"
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+    actions = ["s3:PutObjectTagging"]
+    resources = [
+      "arn:aws:s3:::${aws_s3_bucket.default.id}",
+      "arn:aws:s3:::${aws_s3_bucket.default.id}/*"
+    ]
+    condition {
+      test     = "ForAnyValue:ArnNotEquals"
+      variable = "aws:PrincipalArn"
+      values = [
+        "arn:aws:iam::${local.account_id}:root",
+        "arn:aws:iam::${local.account_id}:assumed-role/${local.malware_iam_role_name}Role/GuardDutyMalwareProtection",
+        module.s3_malware_protection_role[0].arn
+      ]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "s3_malware_protection_assume_role" {
+  count = local.malware_protection
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["malware-protection-plan.guardduty.amazonaws.com"]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [local.account_id]
+    }
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:guardduty:eu-west-1:${local.account_id}:malware-protection-plan/*"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "s3_malware_protection_policy" {
+  count = local.malware_protection
+  statement {
+    sid     = "AllowManagedRuleToSendS3EventsToGuardDuty"
+    effect  = "Allow"
+    actions = ["events:PutRule"]
+    resources = [
+      "arn:aws:events:eu-west-1:${local.account_id}:rule/DO-NOT-DELETE-AmazonGuardDutyMalwareProtectionS3*"
```

Comment:
I cannot find this in the docs but where is this rule coming from? Is this auto created in the account?

eu-west-1 needs to be made dynamic